python/paddle/fluid/imperative/layer_object_helper.py

```diff
@@ -0,0 +1,168 @@
+                                 (dtype, each.dtype))
+        return dtype
+
+    def get_parameter(self, name):
+        param = self.main_program.global_block().var(name)
+        if not isinstance(param, Parameter):
+            raise ValueError("no Parameter name %s found" % name)
+        return param
+
+    def append_bias_op(self,
+                       input_var,
+                       dim_start=1,
+                       dim_end=None,
+                       bias_attr=None):
+        """
+        Append bias operator and return its output. If the user does not set
+        bias_attr, append_bias_op will return input_var
+
+        :param input_var: the input variable. The len(input_var.shape) is
+        larger or equal than 2.
+        :bias_initializer: an instance of a subclass of Initializer used to
+        initialize the bias
+        :param dim_start:
+        :param dim_end: the shape of the bias will be
+        :param bias_attr: the bias_attr of it
+        input_var.shape[dim_start:dim_end]. The bias is broadcasted to other
+        dimensions and added to input_var to get the output
+        """
+        size = list(input_var.shape[dim_start:dim_end])
+        bias_attr = bias_attr
+        if not bias_attr:
+            return input_var
+
+        b = self.create_parameter(
+            attr=bias_attr, shape=size, dtype=input_var.dtype, is_bias=True)
+        tmp = self.create_variable_for_type_inference(dtype=input_var.dtype)
+        self.append_op(
+            type='elementwise_add',
+            inputs={'X': [input_var],
+                    'Y': [b]},
+            outputs={'Out': [tmp]},
+            attrs={'axis': dim_start})
+        return tmp
+
+    def append_activation(self,
+                          input_var,
+                          act=None,
+                          use_cudnn=None,
+                          use_mkl_dnn=None):
+        act = act
+        if act is None:
+            return input_var
+        if isinstance(act, six.string_types):
+            act = {'type': act}
+        else:
+            raise TypeError(str(act) + " should be unicode or str")
+
+        if (use_cudnn is not None) and use_cudnn:
+            act['use_cudnn'] = use_cudnn
+        if (use_mkl_dnn is not None) and use_mkl_dnn:
+            act['use_mkldnn'] = use_mkl_dnn
+        act_type = act.pop('type')
+
+        tmp = input_var
+        # NOTE(dzhwinter): some activation support inplace compution.
+        # NOTE(minqiyang): currently, we don't support inplace in imperative mode
+        if not _in_imperative_mode() and core.IsInplace(act_type):
+            tmp = input_var
+        else:
+            tmp = self.create_variable_for_type_inference(dtype=input_var.dtype)
+        self.append_op(
+            type=act_type,
+            inputs={"X": [input_var]},
+            outputs={"Out": [tmp]},
+            attrs=act)
+        return tmp
+
+    def is_instance(self, param, cls):
```

Comment:
all these method need to be public?

Reply:
this one maybe used by user, maybe we can keep it public for now